exercises/two-bucket/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+#[derive(PartialEq, Eq, Debug)]
+pub enum Bucket {
+    One,
+    Two
+}
+
+/// A struct to hold your results in.
+#[derive(PartialEq, Eq, Debug)]
+pub struct BucketStats {
+    /// The total number of "moves" it should take to reach the desired number of liters, including
+    /// the first fill.
+    pub moves: u8,
+    /// Which bucket should end up with the desired number of liters? (Either "one" or "two")
+    pub goal_bucket: Bucket,
+    /// How many liters are left in the other bucket?
+    pub other_bucket: u8,
+}
+
+/// Solve the bucket problem
+pub fn solve(capacity_1: u8,
+             capacity_2: u8,
+             goal: u8,
+             start_bucket: &str) -> BucketStats
```

Comment:
I think this should be an `&Bucket` also; this prevents the need for a potential panic in your example implementation. It'll mean updating the tests, but that's fine.